cluster.tf

```diff
@@ -1,104 +1,124 @@
-##############################################################################
-# Find valid IKS/ROKS Cluster versions for region
-##############################################################################
-
-data "ibm_container_cluster_versions" "cluster_versions" {}
-
-##############################################################################
-
-
 ##############################################################################
 # Cluster Locals
 ##############################################################################
 
 locals {
-  worker_pools_map = module.dynamic_values.worker_pools_map # Convert list to map
-  clusters_map     = module.dynamic_values.clusters_map     # Convert list to map
-  default_kube_version = {
-    openshift = "${data.ibm_container_cluster_versions.cluster_versions.valid_openshift_versions[length(data.ibm_container_cluster_versions.cluster_versions.valid_openshift_versions) - 1]}_openshift"
-    iks       = data.ibm_container_cluster_versions.cluster_versions.valid_kube_versions[length(data.ibm_container_cluster_versions.cluster_versions.valid_kube_versions) - 1]
-  }
+  workload_cluster = length(module.dynamic_values.clusters_map) >= 1 ? module.dynamic_values.clusters_map["${var.prefix}-workload-cluster"] : null
 }
 
-##############################################################################
-
-
 ##############################################################################
 # Create IKS/ROKS on VPC Cluster
 ##############################################################################
 
-resource "ibm_container_vpc_cluster" "cluster" {
-  for_each          = local.clusters_map
-  name              = "${var.prefix}-${each.value.name}"
-  vpc_id            = each.value.vpc_id
-  resource_group_id = local.resource_groups[each.value.resource_group]
-  flavor            = each.value.machine_type
-  worker_count      = each.value.workers_per_subnet
-  kube_version = (
-    lookup(each.value, "kube_version", null) == "default" # if version is default
-    || lookup(each.value, "kube_version", null) == null   # or if version is null
-    ? local.default_kube_version[each.value.kube_type]    # use default
-    : each.value.kube_version                             # otherwise use value
-  )
-  update_all_workers = lookup(each.value, "update_all_workers", null)
-  tags               = var.tags
-  wait_till          = var.wait_till
-  entitlement        = each.value.entitlement
-  cos_instance_crn   = each.value.cos_instance_crn
-  pod_subnet         = each.value.pod_subnet
-  service_subnet     = each.value.service_subnet
 
-  dynamic "zones" {
-    for_each = each.value.subnets
-    content {
-      subnet_id = zones.value["id"]
-      name      = zones.value["zone"]
-    }
+module "workload_cluster" {
+  depends_on = [
+    module.vpc, module.observability_instances
+  ]
+  count             = length(module.dynamic_values.clusters_map) >= 1 ? 1 : 0
+  source            = "./ocp-all-inclusive"
```

Comment:
I think we should move this module in its own repo, just like what we do internally - so that we can have its own lifecycle, CI, etc independently from the rest of the SLZ module.